src/app/prt-staking/components/prt-section/prt-popup.tsx

```diff
@@ -1,15 +1,18 @@
 import { Modal, ModalBody, ModalContent, ModalOverlay } from '@chakra-ui/react'
 
 import { PrtWidget } from '@/app/prt-staking/components/prt-section/prt-widget'
-import { ProductRevenueToken } from '@/app/prt-staking/types'
+import { usePrtStakingContext } from '@/app/prt-staking/provider'
 
 type Props = {
   isOpen: boolean
   onClose: () => void
-  token: ProductRevenueToken
 }
 
-export function PrtPopup({ isOpen, onClose, token }: Props) {
+export function PrtPopup({ isOpen, onClose }: Props) {
+  const { token } = usePrtStakingContext()
+
+  if (!token) return
```

Comment:
Since it's a specific PRT popup would it be better to just pass the token?

Reply:
Opted for this approach since it makes it easier to move this component somewhere else in the tree if needed and matches our other methods of 'grabbing' this value across these components